src/classes/FeatureParameterMapper_TEST.cls

```diff
@@ -0,0 +1,83 @@
+    * @description Test method to verify that the getInstance method returns a new instance of the 
+    * FeatureParameterMapper class when one does not already exist.
+    ***************************************************************************************************************************/
+    @isTest
+    private static void getInstanceNew() {
+
+        Test.startTest();
+            FeatureParameterMapper featureParamMapperInstance = FeatureParameterMapper.getInstance();
+        Test.stopTest();
+
+        System.assertEquals(FeatureParameterMapper.instance, featureParamMapperInstance, 'Instance of service class returned should match static instance variable.');
+        System.assertEquals(true, featureParamMapperInstance != null, 'Instance of service class should not be null.');
+    }
+
+    /**************************************************************************************************************************
+    * @description Test method to verify that the getInstance method returns the existing instance of the 
+    * FeatureParameterMapper class when one already exists.
+    ***************************************************************************************************************************/
+    @isTest
+    private static void getInstanceExisting() {
+
+        Test.startTest();
+            FeatureParameterMapper featureParamMapperInstance1 = FeatureParameterMapper.getInstance();
+            FeatureParameterMapper featureParamMapperInstance2 = FeatureParameterMapper.getInstance();
+        Test.stopTest();
+
+        System.assertEquals(FeatureParameterMapper.instance, featureParamMapperInstance1, 'Instance of service class returned should match static instance variable.');
+        System.assertEquals(featureParamMapperInstance1, featureParamMapperInstance2, 'Subsequent retrievals of service class instance should return existing instance.');
+        System.assertEquals(true, featureParamMapperInstance1 != null, 'Instance of service class should not be null.');
+    }
+
+    /**************************************************************************************************************************
+    * @description Test method to verify that the getFeatureParameterModel method returns a FeatureParameterModel
+    ***************************************************************************************************************************/
+    @isTest
+    private static void getFeatureParameterModel() {
+
+        Test.startTest();
+            FeatureParameterModel featureParamModel = FeatureParameterMapper.getInstance().getFeatureParameterModel('testFeatureParam');
+        Test.stopTest();
+
+        System.assertEquals('testFeatureParam', featureParamModel.name, 'Feature Parameter Name should be the one which was passed');
+        System.assertEquals(System.now(), ((featureParameterDateModel)featureParamModel).value, 'Feature Parameter value should be returned');
```

Comment:
Likewise here. We may want to make this method specific to an expected FeatureParameterDateModel.
```suggestion
        System.assertEquals(Date.today(), ((featureParameterDateModel)featureParamModel).value, 'Feature Parameter value should be returned');
```